esp-wifi/src/wifi/event.rs

```diff
@@ -0,0 +1,52 @@
+use core::cell::RefCell;
+
+use critical_section::Mutex;
+
+mod sealed {
+    pub trait Sealed {}
+}
+/// The type of handlers of events.
+pub type Handler<T> = dyn FnMut(&T) + Sync + Send;
+// fn default_handler<'a, T>(_: &'a T) {}
+/// Data for a wifi event which can be handled by an event handler.
+pub trait WifiEventData: sealed::Sealed + Sized + 'static {
```

Comment:
This trait is completely unnecessary. You can, in your macro, define a newtype around the event types. That way we could avoid the extremely confusing "static method on C type" situation we have here.

Reply:
I would still need the trait though? How else do you make a static per type? Are you suggesting making a newtype in rust for each `c` type and making that the public interface? I must not understand what you mean, could you clarify?